codes/ruby/chapter_sorting/heap_sort.rb

```diff
@@ -0,0 +1,43 @@
+=begin
+File: heap_sort.rb
+Created Time: 2024-04-10
+Author: junminhong (junminhong1110@gmail.com)
+=end
+
+### 堆的长度为 n ，从节点 i 开始，从顶至底堆化 ###
+def sift_down(nums, n, i)
+  while true
+    # 判断节点 i, l, r 中值最大的节点，记为 ma
+    l = 2 * i + 1
+    r = 2 * i + 2
+    ma = i
+    ma = l if l < n and nums[l] > nums[ma]
+    ma = r if r < n and nums[r] > nums[ma]
```

Comment:
We should use `&&` instead of `and` for boolean expression. The `and` has lower precedence even than the `=`, so we should avoid using it. It is also not recommended in Ruby Style Guide.

```suggestion
    ma = l if l < n && nums[l] > nums[ma]
    ma = r if r < n && nums[r] > nums[ma]
```